engine/dlib/src/dlib/job_thread.cpp

```diff
@@ -207,8 +219,18 @@ void Update(HContext context)
     for(uint32_t i = 0; i < size; ++i)
     {
         JobItem& item = items[i];
-        item.m_Callback(item.m_Context, item.m_Data, item.m_Result);
+        if (item.m_Callback)
+            item.m_Callback(item.m_Context, item.m_Data, item.m_Result);
     }
 }
 
+bool PlatformHasThreadSupport()
+{
+#if defined(DM_HAS_THREADS)
+    return true;
+#else
+    return false;
+#endif
+}
```

Comment:
Maybe this should be moved somewhere? I use it to figure out if we should initialize all of the async job state in the graphics adapters